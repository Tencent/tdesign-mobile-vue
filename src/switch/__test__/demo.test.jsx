/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/switch/demos/base.vue';
import colorVue from '@/switch/demos/color.vue';
import labelVue from '@/switch/demos/label.vue';
import mobileVue from '@/switch/demos/mobile.vue';
import sizeVue from '@/switch/demos/size.vue';
import statusVue from '@/switch/demos/status.vue';

const mapper = {
  baseVue,
  colorVue,
  labelVue,
  mobileVue,
  sizeVue,
  statusVue,
};

describe('Switch', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Switch ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
