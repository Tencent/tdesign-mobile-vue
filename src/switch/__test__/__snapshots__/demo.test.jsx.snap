// Vitest Snapshot v1

exports[`Switch > Switch baseVue demo works fine 1`] = `
<div
  class="t-cell t-cell--middle"
>
  <div
    class="t-cell__left"
  >
    <!---->
    <!---->
  </div>
  <div
    class="t-cell__title"
  >
    基础开关
    <!---->
    <!---->
  </div>
  <!---->
  <div
    class="t-cell__right"
  >
    <div
      class="t-cell__right-icon"
    >
      
      <div
        class="t-switch t-switch--medium t-switch--checked"
      >
        <div
          class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
        >
          <div
            class="t-switch__label t-switch__label--medium t-switch__label--checked"
          >
            <!---->
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Switch > Switch colorVue demo works fine 1`] = `
<div
  class="t-cell t-cell--middle"
>
  <div
    class="t-cell__left"
  >
    <!---->
    <!---->
  </div>
  <div
    class="t-cell__title"
  >
    自定义颜色开关
    <!---->
    <!---->
  </div>
  <!---->
  <div
    class="t-cell__right"
  >
    <div
      class="t-cell__right-icon"
    >
      
      <div
        class="t-switch t-switch--medium t-switch--checked custom-color"
      >
        <div
          class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
        >
          <div
            class="t-switch__label t-switch__label--medium t-switch__label--checked"
          >
            <!---->
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Switch > Switch labelVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="t-cell t-cell--middle"
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      带文字开关
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--medium t-switch--checked"
        >
          <div
            class="t-switch__dot t-switch__dot--medium t-switch__dot--checked"
          >
            <div
              class="t-switch__label t-switch__label--medium t-switch__label--checked"
            >
              开
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="box"
    style="height: 16px;"
  />
  <div
    class="t-cell t-cell--middle"
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      带图标开关
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--medium t-switch--checked"
        >
          <div
            class="t-switch__dot t-switch__dot--medium t-switch__dot--checked"
          >
            <div
              class="t-switch__label t-switch__label--medium t-switch__label--checked"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                style="font-size: 20px;"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M20.99 7.38l-10.61 10.6L4 11.63l1.42-1.41 4.95 4.95 9.2-9.2 1.4 1.42z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Switch > Switch mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Switch 开关
  </h1>
  <p
    class="summary"
  >
    开关用于切换当个设置项的状态，开启、关闭为两个互斥的操作
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        基础开关
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          基础开关
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--checked"
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
              >
                <div
                  class="t-switch__label t-switch__label--medium t-switch__label--checked"
                >
                  <!---->
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        带描述开关
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          带文字开关
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--checked"
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--checked"
              >
                <div
                  class="t-switch__label t-switch__label--medium t-switch__label--checked"
                >
                  开
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="box"
        style="height: 16px;"
      />
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          带图标开关
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--checked"
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--checked"
              >
                <div
                  class="t-switch__label t-switch__label--medium t-switch__label--checked"
                >
                  <svg
                    class="t-icon t-icon-check"
                    fill="none"
                    height="1em"
                    style="font-size: 20px;"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M20.99 7.38l-10.61 10.6L4 11.63l1.42-1.41 4.95 4.95 9.2-9.2 1.4 1.42z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        自定义颜色开关
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          自定义颜色开关
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--checked custom-color"
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
              >
                <div
                  class="t-switch__label t-switch__label--medium t-switch__label--checked"
                >
                  <!---->
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件状态
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        加载状态
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          加载状态
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--disabled"
            >
              <div
                class="t-switch__dot t-switch__dot--medium"
              >
                <div
                  class="t-switch__label t-switch__label--medium"
                >
                  <div
                    class="t-loading"
                    style="font-size: 16.25px;"
                  >
                    <svg
                      class="t-loading__gradient t-icon-loading"
                      height="1em"
                      style="animation-duration: 800ms; width: 16.25px; height: 16.25px;"
                      version="1.1"
                      viewBox="0 0 12 12"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <foreignobject
                        height="12"
                        width="12"
                        x="0"
                        y="0"
                      >
                        <div
                          class="t-loading__gradient-conic"
                        />
                      </foreignobject>
                    </svg>
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          加载状态
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--checked t-switch--disabled"
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--checked"
              >
                <div
                  class="t-switch__label t-switch__label--medium t-switch__label--checked"
                >
                  <div
                    class="t-loading"
                    style="font-size: 16.25px;"
                  >
                    <svg
                      class="t-loading__gradient t-icon-loading"
                      height="1em"
                      style="animation-duration: 800ms; width: 16.25px; height: 16.25px;"
                      version="1.1"
                      viewBox="0 0 12 12"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <foreignobject
                        height="12"
                        width="12"
                        x="0"
                        y="0"
                      >
                        <div
                          class="t-loading__gradient-conic"
                        />
                      </foreignobject>
                    </svg>
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="demo__title"
      >
        禁用状态
      </div>
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          禁用状态
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--disabled"
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--plain"
              >
                <div
                  class="t-switch__label t-switch__label--medium"
                >
                  <!---->
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="t-cell t-cell--middle"
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          禁用状态
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--checked t-switch--disabled"
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
              >
                <div
                  class="t-switch__label t-switch__label--medium t-switch__label--checked"
                >
                  <!---->
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件样式
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        开关尺寸
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      
      <div
        class="t-cell t-cell--middle"
        data-v-8f16ed1c=""
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          大尺寸 32
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--large t-switch--checked"
              data-v-8f16ed1c=""
            >
              <div
                class="t-switch__dot t-switch__dot--large t-switch__dot--checked t-switch__dot--plain"
              >
                <div
                  class="t-switch__label t-switch__label--large t-switch__label--checked"
                >
                  <!---->
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="t-cell t-cell--middle"
        data-v-8f16ed1c=""
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          中尺寸 28
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--medium t-switch--checked"
              data-v-8f16ed1c=""
            >
              <div
                class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
              >
                <div
                  class="t-switch__label t-switch__label--medium t-switch__label--checked"
                >
                  <!---->
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="t-cell t-cell--middle"
        data-v-8f16ed1c=""
      >
        <div
          class="t-cell__left"
        >
          <!---->
          <!---->
        </div>
        <div
          class="t-cell__title"
        >
          小尺寸 24
          <!---->
          <!---->
        </div>
        <!---->
        <div
          class="t-cell__right"
        >
          <div
            class="t-cell__right-icon"
          >
            
            <div
              class="t-switch t-switch--small t-switch--checked"
              data-v-8f16ed1c=""
            >
              <div
                class="t-switch__dot t-switch__dot--small t-switch__dot--checked t-switch__dot--plain"
              >
                <div
                  class="t-switch__label t-switch__label--small t-switch__label--checked"
                >
                  <!---->
                </div>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
</div>
`;

exports[`Switch > Switch sizeVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="t-cell t-cell--middle"
    data-v-8f16ed1c=""
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      大尺寸 32
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--large t-switch--checked"
          data-v-8f16ed1c=""
        >
          <div
            class="t-switch__dot t-switch__dot--large t-switch__dot--checked t-switch__dot--plain"
          >
            <div
              class="t-switch__label t-switch__label--large t-switch__label--checked"
            >
              <!---->
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="t-cell t-cell--middle"
    data-v-8f16ed1c=""
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      中尺寸 28
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--medium t-switch--checked"
          data-v-8f16ed1c=""
        >
          <div
            class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
          >
            <div
              class="t-switch__label t-switch__label--medium t-switch__label--checked"
            >
              <!---->
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="t-cell t-cell--middle"
    data-v-8f16ed1c=""
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      小尺寸 24
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--small t-switch--checked"
          data-v-8f16ed1c=""
        >
          <div
            class="t-switch__dot t-switch__dot--small t-switch__dot--checked t-switch__dot--plain"
          >
            <div
              class="t-switch__label t-switch__label--small t-switch__label--checked"
            >
              <!---->
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Switch > Switch statusVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="t-cell t-cell--middle"
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      加载状态
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--medium t-switch--disabled"
        >
          <div
            class="t-switch__dot t-switch__dot--medium"
          >
            <div
              class="t-switch__label t-switch__label--medium"
            >
              <div
                class="t-loading"
                style="font-size: 16.25px;"
              >
                <svg
                  class="t-loading__gradient t-icon-loading"
                  height="1em"
                  style="animation-duration: 800ms; width: 16.25px; height: 16.25px;"
                  version="1.1"
                  viewBox="0 0 12 12"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <foreignobject
                    height="12"
                    width="12"
                    x="0"
                    y="0"
                  >
                    <div
                      class="t-loading__gradient-conic"
                    />
                  </foreignobject>
                </svg>
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="t-cell t-cell--middle"
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      加载状态
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--medium t-switch--checked t-switch--disabled"
        >
          <div
            class="t-switch__dot t-switch__dot--medium t-switch__dot--checked"
          >
            <div
              class="t-switch__label t-switch__label--medium t-switch__label--checked"
            >
              <div
                class="t-loading"
                style="font-size: 16.25px;"
              >
                <svg
                  class="t-loading__gradient t-icon-loading"
                  height="1em"
                  style="animation-duration: 800ms; width: 16.25px; height: 16.25px;"
                  version="1.1"
                  viewBox="0 0 12 12"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <foreignobject
                    height="12"
                    width="12"
                    x="0"
                    y="0"
                  >
                    <div
                      class="t-loading__gradient-conic"
                    />
                  </foreignobject>
                </svg>
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="demo__title"
  >
    禁用状态
  </div>
  <div
    class="t-cell t-cell--middle"
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      禁用状态
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--medium t-switch--disabled"
        >
          <div
            class="t-switch__dot t-switch__dot--medium t-switch__dot--plain"
          >
            <div
              class="t-switch__label t-switch__label--medium"
            >
              <!---->
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="t-cell t-cell--middle"
  >
    <div
      class="t-cell__left"
    >
      <!---->
      <!---->
    </div>
    <div
      class="t-cell__title"
    >
      禁用状态
      <!---->
      <!---->
    </div>
    <!---->
    <div
      class="t-cell__right"
    >
      <div
        class="t-cell__right-icon"
      >
        
        <div
          class="t-switch t-switch--medium t-switch--checked t-switch--disabled"
        >
          <div
            class="t-switch__dot t-switch__dot--medium t-switch__dot--checked t-switch__dot--plain"
          >
            <div
              class="t-switch__label t-switch__label--medium t-switch__label--checked"
            >
              <!---->
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  
</div>
`;
